apps/dashboard/src/lib/validators/uptime-module.ts

```diff
@@ -0,0 +1,87 @@
+import { BehaviorSubject } from 'rxjs'
+import {
+  calculateApy,
+  getValidatorUptimeSinceDate,
+  uptimePeriodDefinition,
+  type UptimeValue
+} from '@api/utils/entities/component/validator'
+import BigNumber from 'bignumber.js'
+import type { ValidatorCollectionItem } from '@common/gateway-sdk'
+
+type ValidatorAddress = string
+
+export type UptimeData = Partial<
+  Record<UptimeValue, Record<ValidatorAddress, number | undefined>>
+>
+
+export const UptimeModule = () => {
+  const isLoading = new BehaviorSubject<boolean>(false)
```

Comment:
why not a svelte store?

Reply:
I guess svelte store potentially works here as well but it was just arbitrary choice 🤷‍♂️

Reply:
I'd prefer a svelte store here as it is more what we use elsewhere, and it is simpler

Reply:
I think they're both as simple and I'd rather keep it if there are no objections to how it works.

Reply:
Let's not introduce rxjs in dashboard if there is no strong need for it.